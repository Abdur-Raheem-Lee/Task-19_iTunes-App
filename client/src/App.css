body {
    margin: 2%;
    background-image: url(./backgroundImage.jpg);
    background-size: cover;
    background-repeat: no-repeat;
    background-attachment: fixed;
    height: 200px;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
    color: whitesmoke;
}

h1 {
    margin-left: 2%;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.home {
    background: rgba(0, 0, 0, 0.5);
    margin: 1%;
    padding: 1%;
}


/* ------------------------------------- Nav-bar Styling ------------------------------------- */

nav ul h4 {
    float: left;
    display: block;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
    font-size: 17px;
    border-bottom: 3px solid transparent;
}

nav ul h4::after {
    content: '';
    display: block;
    width: 0;
    height: 3px;
    background: skyblue;
    transition: width .3s;
}

nav ul h4:hover::after {
    width: 100%;
}

nav li {
    display: inline-block;
    padding-right: 10px;
}


/* ------------------------------------- Input Styling ------------------------------------- */

h2 {
    margin-left: 1%;
}

input {
    margin-left: 1%;
    border-radius: 3px;
    border: none;
}


/* ------------------------------------- Button Styling ------------------------------------- */

.search-button {
    background-color: none;
    font-size: 24px;
}

.search-button:hover {
    color: yellow;
    cursor: pointer;
}

.search-button .search-tool-tip {
    visibility: hidden;
    width: 120px;
    background: whitesmoke;
    color: black;
    text-align: center;
    padding: 5px;
    border-radius: 6px;
    font-size: 14px;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.search-button:hover .search-tool-tip {
    visibility: visible;
}

.remove-button {
    background-color: none;
    font-size: 24px;
}

.remove-button:hover {
    color: red;
    cursor: pointer;
}

.remove-button .remove-tool-tip {
    visibility: hidden;
    width: 120px;
    background: whitesmoke;
    color: black;
    text-align: center;
    padding: 5px;
    border-radius: 6px;
    font-size: 14px;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.remove-button:hover .remove-tool-tip {
    visibility: visible;
}

.favourite-button {
    background-color: none;
    font-size: 24px;
}

.favourite-button:hover {
    color: green;
    cursor: pointer;
}

.favourite-button .favourite-tool-tip {
    visibility: hidden;
    width: 120px;
    background: whitesmoke;
    color: black;
    text-align: center;
    padding: 5px;
    border-radius: 6px;
    font-size: 14px;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.favourite-button:hover .favourite-tool-tip {
    visibility: visible;
}


/* ------------------------------------- Information Div Styling ------------------------------------- */

.book-info {
    margin: 1%;
    padding: 1%;
}

.song-info {
    margin: 1%;
    padding: 1%;
}

.favourite-info {
    background: rgba(0, 0, 0, 0.5);
    margin: 1%;
    padding: 1%;
}

audio {
    vertical-align: middle;
}

.fa-heart {
    vertical-align: middle;
}