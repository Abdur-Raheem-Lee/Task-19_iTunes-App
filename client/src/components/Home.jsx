import React from "react";

class Home extends React.Component {
  render() {
    return (
      <div class="home">
        <h3>
          <b>About</b>
        </h3>
        <p>This app was created to Find Songs and Books online.</p>
        <p>
          Get started by selecting "Music" and searching songs from any of your
          favourite artists or select "Books" and search books from any of your
          favourite authors!
        </p>
      </div>
    );
  }
}

export default Home;
