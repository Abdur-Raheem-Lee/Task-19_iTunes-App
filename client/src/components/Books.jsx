import React from "react";
import "../App.css";
import parse from "html-react-parser";

class Books extends React.Component {
  constructor() {
    super();
    this.state = {
      dataArr: [],
      bookInput: ""
    };
  }

  //This function gets the dataArr of the ebooks api that will be displayed when the user searches a book
  newSearch() {
    fetch("/books")
      .then(res => res.json())
      .then(dataArr => this.setState({ dataArr }));
  }

  //This async/await function fetches the information from the api
  bookSearch = async () => {
    let search = this.state.bookInput.split(" ").join("+");
    const getBook = await fetch(
      `/book?search=${search}&type=${this.state.type}`
    );
    let res = await getBook.json();
    this.setState({
      dataArr: res
    });
  };

  //This function gets and adds the information of the book the user selected to the favourite Books json file
  favoriteBook = i => {
    let favPic = {
      id: i.trackId,
      author: i.artistName,
      artwork: i.artworkUrl100,
      book: i.trackName,
      description: i.description
    };

    //This fetch method  is used to get and post the information of the books to "favouriteBooks" json file
    fetch("/favoritesBooks", {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(favPic)
    });
    alert(i.trackName + " was added to Favorites");
  };

  render() {
    return (
      <div>
        <h1>E-Books</h1>
        <input
          type="text"
          onChange={e => this.setState({ bookInput: e.target.value })}
          input
        />
        &emsp;
        <i onClick={() => this.bookSearch()} class="search-button fa fa-search">
          {" "}
          &nbsp;
          <span class="search-tool-tip">Search</span>
        </i>
        {/*This div displays all the information of books related to what was entered into the search bar*/}
        {this.state.dataArr.map(sortArr => (
          <article key={sortArr.trackId}>
            <h4>
              {sortArr.artistName} &nbsp; - &nbsp;
              {sortArr.trackName}
            </h4>
            {parse(`<p>${sortArr.description}</p>`)}
            <img src={sortArr.artworkUrl100} alt="Loading..." />
            &emsp;
            <i
              onClick={() => {
                this.favoriteBook(sortArr);
              }}
              class="favourite-button fa fa-heart"
            >
              {" "}
              &nbsp;
              <span class="favourite-tool-tip">Add to Favourites</span>
            </i>
            <hr />
          </article>
        ))}
      </div>
    );
  }
}

export default Books;
