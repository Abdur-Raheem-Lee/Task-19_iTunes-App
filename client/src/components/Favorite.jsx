import React from "react";
import "../App.css";
import parse from "html-react-parser";

class Favorite extends React.Component {
  constructor() {
    super();
    this.state = {
      favSongs: [],
      favBooks: []
    };
  }

  //This function fetches the information from the favourites Music json file.
  componentDidMount() {
    fetch("/favoritesMusic")
      .then(res => res.json())
      .then(music => this.setState({ favSongs: music }));

    fetch("/favoritesBooks")
      .then(res => res.json())
      .then(books => this.setState({ favBooks: books }));
  }

  //This function removes the song selected by the user to be removed
  songRemove = i => {
    let songDeleteFromFav = {
      deleted: i.id
    };

    //This fetches the updated data from removed favourites
    fetch("/favoritesMusic", {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(songDeleteFromFav)
    });

    //This refreshes the page when the user removes a favourite so that the user can see that the entry was deleted
    document.location.reload();
  };

  //This function removes the book selected by the user to be removed
  bookRemove = i => {
    let bookRemovedFromFav = {
      deleted: i.id
    };

    //This fetches the updated data from removed favourites
    fetch("/favoritesBooks", {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(bookRemovedFromFav)
    });

    //This refreshes the page when the user removes a favourite so that the user can see that the entry was deleted
    document.location.reload();
  };

  render() {
    return (
      <div>
        <div class="favourite-info">
          <h3>
            <u>Below are all your favourite Songs:</u>
          </h3>

          {/*This div displays all the songs the user favourited*/}
          {this.state.favSongs.map(sortMusic => (
            <article key={sortMusic.trackId}>
              <h4>
                {sortMusic.artist} &nbsp; - &nbsp; {sortMusic.track}
              </h4>
              <img src={sortMusic.artwork} alt="Loading..." /> &emsp;
              <i
                onClick={() => {
                  this.songRemove(sortMusic);
                }}
                class="remove-button fa fa-trash"
              >
                &nbsp;
                <span class="remove-tool-tip">Remove from Favourites</span>
              </i>
            </article>
          ))}
        </div>
        <div class="favourite-info">
          <h3>
            <u>Below are all your favourite Books:</u>
          </h3>

          {/*This div displays all the books the user favourited*/}
          {this.state.favBooks.map(sortBooks => (
            <article key={sortBooks.artwork}>
              <h4>
                {sortBooks.artist} &nbsp; - &nbsp; {sortBooks.track}
              </h4>
              <img src={sortBooks.artwork} alt="Loading..." />
              &emsp;
              <i
                onClick={() => {
                  this.bookRemove(sortBooks);
                }}
                class="remove-button fa fa-trash"
              >
                &nbsp;
                <span class="remove-tool-tip">Remove from Favourites</span>
              </i>
              <br />
              {parse(`<p>${sortBooks.description}</p>`)}{" "}
              <p>{sortBooks.track}</p>
            </article>
          ))}
        </div>
      </div>
    );
  }
}

export default Favorite;
