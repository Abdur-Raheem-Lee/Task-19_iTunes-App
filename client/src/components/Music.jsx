import React from "react";
import "../App.css";
import parse from "html-react-parser";

class Music extends React.Component {
  constructor() {
    super();
    this.state = {
      dataArr: [],
      songInput: ""
    };
  }

  //This function gets the dataArr of the ebooks api that will be displayed when the user searches a track
  newSearch() {
    fetch("/music")
      .then(res => res.json())
      .then(dataArr => this.setState({ dataArr }));
  }

  //This async/await function fetches the information from the api
  songSearch = async () => {
    let search = this.state.songInput.split(" ").join("+");
    const getBook = await fetch(
      `/music?search=${search}&type=${this.state.type}`
    );
    let res = await getBook.json();
    this.setState({
      dataArr: res
    });
  };

  //This function gets and adds the information of the track the user selected to the favourite Music json file
  favoriteSong = i => {
    let favPic = {
      id: i.trackId,
      artist: i.artistName,
      artwork: i.artworkUrl100,
      track: i.trackName,
      description: i.description
    };

    //This fetch method  is used to get and post the information of the books to "favouriteMusic" json file
    fetch("/favoritesMusic", {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(favPic)
    });
    alert(i.trackName + " was added to Favorites");
  };

  render() {
    return (
      <div>
        <h1>Tunes</h1>
        <input
          type="text"
          onChange={e => this.setState({ songInput: e.target.value })}
          input
        />
        &emsp;
        <i onClick={() => this.songSearch()} class="search-button fa fa-search">
          {" "}
          &nbsp;
          <span class="search-tool-tip">Search</span>
        </i>
        {/*This div displays all the information of books related to what was entered into the search bar*/}
        {this.state.dataArr.map(sortArr => (
          <article key={sortArr.trackId}>
            <h4>
              {sortArr.artistName} &nbsp; - &nbsp;
              {sortArr.trackName}
            </h4>
            <img src={sortArr.artworkUrl100} alt="Loading..." />
            &emsp;
            <i
              onClick={() => {
                this.favoriteSong(sortArr);
              }}
              class="favourite-button fa fa-heart"
            >
              {" "}
              &nbsp;
              <span class="favourite-tool-tip">Add to Favourites</span>
            </i>
            <hr />
          </article>
        ))}
      </div>
    );
  }
}

export default Music;
