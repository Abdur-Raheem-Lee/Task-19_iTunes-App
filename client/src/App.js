import React from "react";
import "./App.css";
import { HashRouter as Router, Switch, Route, Link } from "react-router-dom";
import Home from "./components/Home";
import Music from "./components/Music";
import Books from "./components/Books";
import Favorite from "./components/Favorite";

class App extends React.Component {
  render() {
    return (
      // Using "Router", "Switch", "Route" and "Link" to navigate between components so that the webpage
      // doesn't need to load everytime the user goes to a new component/page
      <Router>
        <div>
          <h1>The Music and eBook Store</h1>
          <nav>
            <ul>
              <li>
                <Link to={"/Home"}>
                  <h4>
                    <b>Home</b>
                  </h4>
                </Link>
              </li>
              <li>
                <Link to={"/Music"}>
                  <h4>
                    <b>Music</b>
                  </h4>
                </Link>
              </li>
              <li>
                <Link to={"/Books"}>
                  <h4>
                    <b>Books</b>
                  </h4>
                </Link>
              </li>
              <li>
                <Link to={"/Favorite"}>
                  <h4>
                    <b>Favorites</b>
                  </h4>
                </Link>
              </li>
            </ul>
          </nav>
          <Switch>
            <Route path="/Home" component={Home} />
            <Route path="/Music" component={Music} />
            <Route path="/Books" component={Books} />
            <Route path="/Favorite" component={Favorite} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
