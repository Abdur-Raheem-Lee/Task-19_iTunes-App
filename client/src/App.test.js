import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App from "./App";
import Books from "./App";
import { getBook, res } from "./components/Books";

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render( < App / > , div);
    ReactDOM.unmountComponentAtNode(div);
});

// Testing fetch()
it("Testing Api", () => {
    done => {
        function callBack(data) {
            expect(data).toBe(res);
            done();
        }
        callBack();
        const div = document.createElement("div");
        ReactDOM.renderer( < Book / > , div);
        ReactDOM.unmountComponentAtNode(div);
    };
});

// Snapshot Testing
it("The test works!", () => {
    const component = renderer.create( < Books / > );
    let app = component.toJSON();
    expect(app).toMatchSnapshot();
});