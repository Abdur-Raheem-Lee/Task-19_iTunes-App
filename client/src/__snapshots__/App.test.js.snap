// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The test works! 1`] = `
<div>
  <h1>
    The Music and eBook Store
  </h1>
  <nav>
    <ul>
      <li>
        <a
          href="#/Home"
          onClick={[Function]}
        >
          <h4>
            <b>
              Home
            </b>
          </h4>
        </a>
      </li>
      <li>
        <a
          href="#/Music"
          onClick={[Function]}
        >
          <h4>
            <b>
              Music
            </b>
          </h4>
        </a>
      </li>
      <li>
        <a
          href="#/Books"
          onClick={[Function]}
        >
          <h4>
            <b>
              Books
            </b>
          </h4>
        </a>
      </li>
      <li>
        <a
          href="#/Favorite"
          onClick={[Function]}
        >
          <h4>
            <b>
              Favorites
            </b>
          </h4>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <h1>
    E-Books
  </h1>
  <input
    input={true}
    onChange={[Function]}
    type="text"
  />
   
  <i
    class="search-button fa fa-search"
    onClick={[Function]}
  >
     
     
    <span
      class="search-tool-tip"
    >
      Search
    </span>
  </i>
</div>
`;
